packages/network-controller/tests/provider-api-tests/block-hash-in-response.ts

```diff
@@ -1,9 +1,13 @@
+import { ConstantBackoff } from '@metamask/controller-utils';
```

Comment:
Lots of changes here, but we are ultimately testing:

that we can pass getRpcServiceOptions to NetworkController, and this affects the retry and circuit breaker behavior for individual RPC endpoints
that the new events that we've added get called
And if it looks like we are repeating the same test over and over... we are. These tests seriously need to be refactored :(